File: backend/dotnet-services/OrderService/Shared/Events/StockUpdateEvent.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Shared.Events
{
    public class StockUpdateEvent
    {
        public string EventType { get; set; } // "StockDecrease" or "StockRestock"
        public long BranchId { get; set; }
        public List<StockUpdateItem> Items { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public string Reference { get; set; }
        public long? OrderId { get; set; }
        public long RefundId { get; set; }
    }
}

File: backend/springboot-services/inventory-service/src/main/java/com/market_os/inventory_service/dto/StockUpdateEvent.java
package com.market_os.inventory_service.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class StockUpdateEvent {
    
    private String eventType; // "StockDecrease" or "StockRestock"
    private Long branchId;
    private List<StockUpdateItem> items;
    
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime createdAt;
    
    private String reference;
    private Long orderId;
    private Long refundId;
} 